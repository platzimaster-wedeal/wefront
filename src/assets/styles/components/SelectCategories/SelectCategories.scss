@import '../../partials/sizes.scss';
@import '../../partials/colors.scss';
@import '../../partials/shadows.scss';
@import '../../Breakpoints.scss';

.select-categories {
  position: relative;
  width: 100%;
  display: flex;
  flex-flow: column nowrap;

   &__title {
    @include font-size($font-size-normal);
    font-weight: 500;
    color: $color-black-90;
    padding: $space-size-extra-small 0;
  }

  &__select-list {
    width: 100%;
    display: flex;
    flex-flow: row nowrap;
    overflow-y: scroll;

    & li {
      margin: $space-size-small $space-size-small $space-size-small 0;
      cursor: pointer;
    }
  }

  &__list {
    width: 100%;
    display: flex;
    flex-flow: row nowrap;
    overflow-y: scroll;

    & li {
      position: relative;
      margin: $space-size-small $space-size-small $space-size-small 0;
      cursor: pointer;
    }

    &--close {
      &::before{
        content: 'X';
        display: flex;
        align-items: center;
        justify-content: center;
        position: absolute;
        width: 21px;
        height: 21px;
        background-color: $color-background;
        border: 1px solid $color-primary;
        border-radius: 50%;
        opacity: 0;
        right: -6%;
        top: -12%;  
        transition: .3s ease-in-out;
        box-shadow: $shadow-normal;
      }
    }

    &--close:hover {
      &::before {
        opacity: 1;
      }
    }

  }
}