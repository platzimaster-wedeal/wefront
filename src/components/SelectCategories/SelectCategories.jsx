import React, { useState } from "react";

// Hooks
import { useInputForm } from "../../hooks/useInputForm/useInputForm";

// Styles
import "../../assets/styles/components/SelectCategories/SelectCategories.scss";

// Components
import Labels from "../Labels/Labels";

const SelectCategories = ({
 categories = ["art", "science", "magician"],
 userCategories = [],
}) => {
 // States
 const [currentCategories, setCurrentCategories] = useState(userCategories);
 const [categoryOptions, setCategoryOptions] = useState(categories);
 const [filterCategory, setFilterCategory] = useState(categories);

 // Handle the selection of categories for the user
 const handleSelectCategory = (category) => {
  if (currentCategories.includes(category)) return null;

  setCurrentCategories((prevs) => [...prevs, category]);
 };

 // Handle the filter of the categories, for use find the skill or category that need
 const handleFilterCategory = (ev) => {
  ev.preventDefault();
  let filterCategories = categoryOptions.filter((category) =>
   category.includes(ev.currentTarget.value)
  );
  setFilterCategory(filterCategories);
 };

 // Handle the event of delete some selected category of user
 const handleDeleteCategory = (category) => {
  const newCurrentCategories = currentCategories.filter(
   (currCategory) => currCategory != category
  );
  setCurrentCategories(newCurrentCategories);
 };

 return (
  <div className="select-categories">
   <label htmlFor="categories" className="select-categories__title">
    Categories
   </label>
   <input
    type="text"
    placeholder="Search and select the skills that you want"
    id="categories"
    onChange={handleFilterCategory}
   />

   <ul className="select-categories__select-list">
    {filterCategory &&
     filterCategory.map((category) => (
      <li key={category} onClick={() => handleSelectCategory(category)}>
       {currentCategories.includes(category) ? (
        <Labels name={category} active />
       ) : (
        <Labels name={category} />
       )}
      </li>
     ))}
   </ul>

   <small className="select-categories__title">Selected Categories</small>
   <ul className="select-categories__list">
    {currentCategories && currentCategories.length > 0 ? (
     currentCategories.map((category) => (
      <li
       key={category}
       className="select-categories__list--close"
       onClick={() => handleDeleteCategory(category)}>
       {" "}
       <Labels name={category} active />{" "}
      </li>
     ))
    ) : (
     <li>There is no categories selected.</li>
    )}
   </ul>
  </div>
 );
};

export default SelectCategories;
