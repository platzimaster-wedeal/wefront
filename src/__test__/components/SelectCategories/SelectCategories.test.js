import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import SelectCategories from "../../../components/SelectCategories/SelectCategories";

describe("Tests for SelectCategories components", () => {
 const SelectCategoriesMounted = mount(<SelectCategories />);
 const SelectCategoriesCreated = create(<SelectCategories />);

 describe("Visual tests for the SelectCategoires component", () => {
  test("should the component render one component", () => {
   expect(SelectCategoriesMounted.length).toEqual(1);
  });
  test("should the component have the same JSON tree", () => {
   expect(SelectCategoriesCreated.toJSON).toMatchSnapshot();
  });
  test("should the title of the component be a string", () => {
   expect(
    typeof SelectCategoriesMounted.find(".select-categories label").text()
   ).toEqual("string");
  });

  test("should the second title of the component be a string", () => {
   expect(
    typeof SelectCategoriesMounted.find(".select-categories small").text()
   ).toEqual("string");
  });
 });
});
