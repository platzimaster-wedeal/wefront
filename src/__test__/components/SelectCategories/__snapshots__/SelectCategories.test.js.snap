// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for SelectCategories components Visual tests for the SelectCategoires component should the component have the same JSON tree 1`] = `[Function]`;
